src/content/actions/receive-profile.js

```diff
@@ -232,19 +234,80 @@ export function receiveProfileFromWeb(profile: Profile): ThunkAction {
   };
 }
 
-export function errorReceivingProfileFromWeb(error: any): Action {
+export function temporaryErrorReceivingProfileFromWeb(error: TemporaryError): Action {
   return {
-    type: 'ERROR_RECEIVING_PROFILE_FROM_WEB',
+    type: 'TEMPORARY_ERROR_RECEIVING_PROFILE_FROM_WEB',
     error,
   };
 }
 
+export function fatalErrorReceivingProfileFromWeb(error: Error): Action {
+  return {
+    type: 'FATAL_ERROR_RECEIVING_PROFILE_FROM_WEB',
+    error,
+  };
+}
+
+function wait(delayMs) {
+  return new Promise(resolve => setTimeout(resolve, delayMs));
+}
+
+type FetchProfileArgs = {
+  url: string,
+  onTemporaryError: TemporaryError => void,
+};
+/**
+ * Tries to fetch a profile on URL url. If the profile is not found,
+ * onTemporaryError is called with an appropriate error, we wait 1 second, and
+ * then tries again. If we still can't find the profile after 11 tries, we throw
+ * a permanent error.
+ * @param {string} url URL to fetch
+ * @param {Function} onTemporaryError callback that will be called for
+ * tempoerary errors.
+ * @returns {Promise.<String>} Resolves to the text profile if successful,
+ * rejects with an error for fatal errors.
+ */
+async function fetchProfile({ url, onTemporaryError }: FetchProfileArgs) {
+  const MAX_WAIT_SECONDS = 11; // 11 tries for 10 seconds
+
+  for (let i = 0; i < MAX_WAIT_SECONDS; i++) {
+    const response = await fetch(url);
+    if (response.ok) {
+      const text = await response.text();
+      return text;
+    }
+
+    if (response.status !== 404) {
+      throw new Error(oneLine`
+        Could not fetch the profile on remote server.
+        Response was: ${response.status} ${response.statusText}.
+      `);
+    }
+
+    onTemporaryError(new TemporaryError(
+      'Profile not found on remote server.',
```

Comment:
Here it's really the structured error, we have all the information needed for the UI in the structure. So I'd rather not change it here. Or maybe add a toString method to this error?

I'd be happy to change what's displayed to the user though.